ui/src/tool-panel/operational-point/operational-point-switches-infobox.tsx

```diff
@@ -0,0 +1,430 @@
+            contentVisible={contentVisible}
+            onContentVisibilityChange={() => onVisibilityChange('switches')}>
+            {isInitializing ? (
+                <Spinner />
+            ) : (
+                <>
+                    <OperationalPointSwitchesDirectionInfobox
+                        layoutContext={layoutContext}
+                        operationalPoint={operationalPoint}
+                        changeTimes={changeTimes}
+                        switches={linkedSwitches}
+                        linkingAction={unlinkSwitches}
+                        linkingDirection={'unlinking'}
+                    />
+                    <OperationalPointSwitchesDirectionInfobox
+                        layoutContext={layoutContext}
+                        operationalPoint={operationalPoint}
+                        changeTimes={changeTimes}
+                        switches={unlinkedSwitches}
+                        linkingAction={linkSwitches}
+                        linkingDirection={'linking'}
+                    />
+                </>
+            )}
+        </Infobox>
+    );
+};
+
+function useLinkingSwitches(
+    layoutContext: LayoutContext,
+    operationalPoint: OperationalPoint,
+    changeTimes: ChangeTimes,
+): {
+    isInitializing: boolean;
+    linkedSwitches: { switchItem: LayoutSwitch; polygonInclusions: OperationalPointId[] }[];
+    unlinkedSwitches: { switchItem: LayoutSwitch; polygonInclusions: OperationalPointId[] }[];
+    linkSwitches: (switches: { id: LayoutSwitchId; name: string }[]) => void;
+    unlinkSwitches: (switches: { id: LayoutSwitchId; name: string }[]) => void;
+} {
+    const { t } = useTranslation();
+
+    const [isInitializing, setIsInitializing] = React.useState(true);
+    const [linkingInFlight, addLinkingInFlight, deleteFromLinkingInFlight] =
+        useSetState<LayoutSwitchId>();
+    const [unlinkingInFlight, addUnlinkingInFlight, deleteFromUnlinkingInFlight] =
+        useSetState<LayoutSwitchId>();
+
+    const [preliminaryLinked, addPreliminaryLinked, deletePreliminaryLinked, setPreliminaryLinked] =
+        useSetState<LayoutSwitchId>();
+    const [
+        preliminaryUnlinked,
+        addPreliminaryUnlinked,
+        deletePreliminaryUnlinked,
+        setPreliminaryUnlinked,
+    ] = useSetState<LayoutSwitchId>();
+
+    const [operationalPointSwitches, setOperationalPointSwitches] = React.useState<
+        SwitchWithOperationalPointPolygonInclusions[]
+    >([]);
+
+    const [switchItems, setSwitchItems] = React.useState<LayoutSwitch[]>([]);
+    useRateLimitedTwoPartEffect(
+        async () => {
+            const loadedOperationalPointSwitches = await findOperationalPointSwitches(
+                layoutContext,
+                operationalPoint.id,
+            );
+            const loadedSwitchItems = await getSwitches(
+                loadedOperationalPointSwitches.map(({ switchId }) => switchId),
+                layoutContext,
+            );
```

Comment:
Rupesin miettimään että kannattaisikohan `findOperationalPointSwitches()`:in palauttaa koko vaihdeolio pelkän id:n palauttamisen sijaan. Tässä kuitenkin halutaan koko vaihdeoliot heti ja molempien tulokset kuitenkin kakutetaan, niin säästyttäisiin toisen kutsun tekemiseltä. Se hyvä puoli tässä mallissa toki on, että jos tuo ensimmäinen kutsu palauttaa jotain sellaisia vaihteita joita ei ole kakussa, niin tuo jälkimmäinen kutsu käy ne hakemassa kakkuun muillekin tarjolle